sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -168,5 +172,73 @@ internal static string GetOperationName(Activity activity, ref AzMonList MappedT
 
             return activity.DisplayName;
         }
+
+        private static void AddExceptionTelemetryFromActivityExceptionEvents(Activity activity, ref TagEnumerationState monitorTags, string roleName, string roleInstance, string instrumentationKey, List<TelemetryItem> telemetryItems)
+        {
+            if (!activity.Events.Any())
+            {
+                return;
+            }
+
+            foreach (var evnt in activity.Events)
+            {
+                if (evnt.Name == SemanticConventions.AttributeExceptionEventName)
+                {
+                    try
+                    {
+                        var exceptionTelemetryItem = new TelemetryItem("Exception", activity, ref monitorTags, roleName, roleInstance, instrumentationKey);
+                        SetExceptionDataDetailsOnTelemetryItem(evnt.Tags, exceptionTelemetryItem);
+                        telemetryItems.Add(exceptionTelemetryItem);
+                    }
+                    catch (Exception ex)
+                    {
+                        AzureMonitorExporterEventSource.Log.WriteError("FailedToExtractExceptionFromActivityEvent", ex);
+                    }
+                }
+            }
+        }
+
+        internal static void SetExceptionDataDetailsOnTelemetryItem(IEnumerable<KeyValuePair<string, object>> activityEventTags, TelemetryItem exceptionTelemetryItem)
+        {
+            string exceptionType = null;
+            string exceptionStackTrace = null;
+            string exceptionMessage = null;
+
+            // TODO: update to use perf improvements in .NET7.0
+            foreach (var tag in activityEventTags)
+            {
+                if (tag.Key == SemanticConventions.AttributeExceptionType)
+                {
+                    exceptionType = tag.Value.ToString();
+                    continue;
+                }
+                if (tag.Key == SemanticConventions.AttributeExceptionMessage)
+                {
+                    exceptionMessage = tag.Value.ToString();
+                    continue;
+                }
+                if (tag.Key == SemanticConventions.AttributeExceptionStacktrace)
+                {
+                    exceptionStackTrace = tag.Value.ToString();
+                    continue;
+                }
+            }
+
+            TelemetryExceptionDetails exceptionDetails = new(exceptionMessage);
```

Comment:
is there a guarantee that exceptionMessage,Type,Stack will be non null here?

Reply:
No. type and stack can be null. Message cannot be otherwise it will throw. Refactored the code a bit to save extra work.